### PR TITLE
Iss12

### PR DESCRIPTION
**Pull request by 4-life**
Filtering state https://github.com/AdguardTeam/AdguardAssistant/issues/12

### CODE REVIEW

File: src/controllers/mainMenuController.js [hunk 1]
```diff
@@ -8,30 +8,30 @@
  * @constructor
  */
 /* global Ioc, StringUtils, punycode */
-var DetailedMenuController = function ($, wot, localization, gmApi, settings) { // jshint ignore:line
+var DetailedMenuController = function($, wot, localization, gmApi, settings) { // jshint ignore:line
     var contentDocument = null;
     var iframeCtrl = null;
     var domain = null;
+    var FILTERING_STATE = 'Adguard_Filtering_State';
```
Comment (atropnikov) [inline src/controllers/mainMenuController.js]: It had better rename to something more clear `FILTERING_STATE_LS_PROPERTY`

File: src/controllers/mainMenuController.js [hunk 2]
```diff
@@ -42,44 +42,83 @@ var DetailedMenuController = function ($, wot, localization, gmApi, settings) {
...
 
-    var doNotBlock = function () {
-        gmApi.ADG_temporaryDontBlock(30, function () {
+    /**
+     * Storing the filtering state for quick initialization
+     *
+     * @param {Boolean} state  on/off filtering state
+     */
+    var setFilteringStateToStore = function(state) {
+        localStorage.setItem(FILTERING_STATE, JSON.stringify({
```
Comment (atropnikov) [inline src/controllers/mainMenuController.js]: I would wrap it into try catch block, just in case.

File: src/controllers/mainMenuController.js [hunk 3]
```diff
@@ -42,44 +42,83 @@ var DetailedMenuController = function ($, wot, localization, gmApi, settings) {
...
-    var setInitFilteringState = function () {
-        gmApi.ADG_isFiltered(function (isFiltered) {
-            var input = contentDocument.getElementById('is-filter');
+    var setInitFilteringState = function() {
+        var input = contentDocument.getElementById('is-filter');
+        input.checked = getFilteringStateFromStore();
+        gmApi.ADG_isFiltered(function(isFiltered) {
             input.checked = isFiltered;
+            setFilteringStateToStore(isFiltered);
+            showHideBlockAdButton(isFiltered);
```
Comment (atropnikov) [inline src/controllers/mainMenuController.js]: Why don't you call `showHideBlockAdButton` immediately after you've read filtering state from the local storage?

Comment (4-life) [inline src/controllers/mainMenuController.js]: @atropnikov isFiltered may be another value after app responsed

Comment (atropnikov) [inline src/controllers/mainMenuController.js]: I meant "add" not  "replace". Won't the button flicker?

Comment (4-life) [inline src/controllers/mainMenuController.js]: no it does. This button not rendering before app responsed

Comment (atropnikov) [inline src/controllers/mainMenuController.js]: Ok.

File: src/controllers/mainMenuController.js [hunk 4]
```diff
@@ -42,44 +42,87 @@ var DetailedMenuController = function ($, wot, localization, gmApi, settings) {
...
+     *
+     * @param {Boolean} state  on/off filtering state
+     */
+    var setFilteringStateToStore = function(state) {
+        try {
+            localStorage.setItem(FILTERING_STATE_LS_PROPERTY, JSON.stringify({
+                "state": state
+            }));
+        } catch (ex) {
+            return null;
```
Comment (ameshkov) [inline src/controllers/mainMenuController.js]: Logging?

File: src/controllers/mainMenuController.js [hunk 5]
```diff
@@ -42,44 +42,87 @@ var DetailedMenuController = function ($, wot, localization, gmApi, settings) {
...
+    var getFilteringStateFromStore = function() {
+        try {
+            var state = localStorage.getItem(FILTERING_STATE_LS_PROPERTY);
+            if (state) {
+                return JSON.parse(state).state;
+            } else {
+                return false;
+            }
+        } catch (ex) {
+            return false;
```
Comment (ameshkov) [inline src/controllers/mainMenuController.js]: Logging?

File: src/controllers/mainMenuController.js [hunk 6]
```diff
@@ -8,30 +8,30 @@
  * @constructor
  */
 /* global Ioc, StringUtils, punycode */
-var DetailedMenuController = function ($, wot, localization, gmApi, settings) { // jshint ignore:line
+var DetailedMenuController = function($, wot, localization, gmApi, settings) { // jshint ignore:line
     var contentDocument = null;
     var iframeCtrl = null;
     var domain = null;
+    var FILTERING_STATE_LS_PROPERTY = 'Adguard_Filtering_State';
```
Comment (ameshkov) [inline src/controllers/mainMenuController.js]: It'd be better to have some generic name like `__agfstate` just to not highlight ourselves in the local storage.

Comment (4-life) [inline src/controllers/mainMenuController.js]: also button position storing with key "Adguard_Button_Position"

Comment (ameshkov) [inline src/controllers/mainMenuController.js]: Makes sense to rename it as well

**Pull request merged**